lib/iris/util.py

```diff
@@ -43,52 +43,6 @@
 import iris.exceptions
 
 
-def broadcast_weights(weights, array, dims):
-    """
-    Broadcast a weights array to the shape of another array.
-
-    Each dimension of the weights array must correspond to a dimension
-    of the other array.
-
-    .. deprecated:: 1.6
-
-       Please use :func:`~iris.util.broadcast_to_shape()`.
-
-    Args:
-
-    * weights (:class:`numpy.ndarray`-like):
-        An array of weights to broadcast.
-
-    * array (:class:`numpy.ndarray`-like):
-        An array whose shape is the target shape for *weights*.
-
-    * dims (:class:`list` :class:`tuple` etc.):
-        A sequence of dimension indices, specifying which dimensions of
-        *array* are represented in *weights*. The order the dimensions
-        are given in is not important, but the order of the dimensions
-        in *weights* should be the same as the relative ordering of the
-        corresponding dimensions in *array*. For example, if *array* is
-        4d with dimensions (ntime, nlev, nlat, nlon) and *weights*
-        provides latitude-longitude grid weightings then *dims* could be
-        set to [2, 3] or [3, 2] but *weights* must have shape
-        (nlat, nlon) since the latitude dimension comes before the
-        longitude dimension in *array*.
-
-    """
-    warn_deprecated('broadcast_weights() is deprecated and will be removed '
```

Comment:
@lbdreyer you will also need to remove the import for this function on [L41](https://github.com/SciTools/iris/pull/2686/files#diff-b6f4583aa861d9305fec9d55d9c18dfaL41).